File: Trees/Medium/1305/1305.py
from typing import List

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> List[int]:
        all_tree_ints = []
        def traversal(node: TreeNode, outputList: List):
            if node is not None:
                traversal(node.left, all_tree_ints)
                outputList.append(node.val)
                traversal(node.right, all_tree_ints)
        traversal(root1, all_tree_ints)
        traversal(root2, all_tree_ints)
        all_tree_ints.sort()
        return all_tree_ints

File: Trees/Medium/1305/1305.c
#include <stdio.h>
#include <stdlib.h>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

void traversal(struct TreeNode* node, int* arr, int* idx){
    if(node != NULL){
        traversal(node->left, arr, idx);
        arr[(*idx)++] = node->val;
        traversal(node->right, arr, idx);
    }
}

int* getAllElements(struct TreeNode* root1, struct TreeNode* root2, int* returnSize){
    int arr1[5000], arr2[5000];
    int idx1 = 0, idx2 = 0;

    traversal(root1, arr1, &idx1);
    traversal(root2, arr2, &idx2);

    int *result = (int *)malloc(sizeof(int) * (idx1 + idx2));
    *returnSize = 0;

    int i = 0, j = 0;
    while(i < idx1 && j < idx2){
        if(arr1[i] <= arr2[j]){
            result[(*returnSize)++] = arr1[i++];
        }else{
            result[(*returnSize)++] = arr2[j++];
        }
    }

    while(i < idx1){
        result[(*returnSize)++] = arr1[i++];
    }
    while(j < idx2){
        result[(*returnSize)++] = arr2[j++];
    }
    free(arr1);
    free(arr2);
    return result;
}